checkov/circleci_pipelines/checks/prevent_development_orbs.py

```diff
@@ -0,0 +1,34 @@
+from __future__ import annotations
+from typing import Any
+
+from checkov.circleci_pipelines.base_circleci_pipelines_check import BaseCircleCIPipelinesCheck
+from checkov.common.models.enums import CheckResult
+from checkov.yaml_doc.enums import BlockType
+
+class PreventDevelopmentOrbs(BaseCircleCIPipelinesCheck):
+    def __init__(self) -> None:
+        name = "Ensure mutable development orbs are not used."
+        id = "CKV_CIRCLECIPIPELINES_3"
+        super().__init__(
+            name=name,
+            id=id,
+            block_type=BlockType.ARRAY,
+            supported_entities=["orbs.{orbs: @}"]
+        )
+
+    def scan_conf(self, conf: dict[str, Any]) -> tuple[CheckResult, dict[str, Any]]:
+        badOrbInBlock = False
+        for orb in conf:
+            if type(conf[orb]) == str:
+                # Special __ vars show up in this dict too.
+                if "@dev" in conf[orb]:
+                    badOrbInBlock = True
+                    # We only get one return per orb: section, regardless of how many orbs, so set a flag and error later.
+                    # Potentially more JMEpath reflection-foo can resolve this so we end up with a call to scan_entity_conf per orb.
+        
+        if badOrbInBlock:
+            return CheckResult.FAILED, conf
+        else:
+            return CheckResult.PASSED, conf
```

Comment:
```suggestion
        for orb in conf.values():
            if isinstance(orb, str):
                # Special __ vars show up in this dict too.
                if "@dev" in orb:
                    # We only get one return per orb: section, regardless of how many orbs, so set a flag and error later.
                    # Potentially more JMEpath reflection-foo can resolve this so we end up with a call to scan_entity_conf per orb.
                    return CheckResult.FAILED, conf
        
        return CheckResult.PASSED, conf
```
actually there is no need for the flag and you can just iterate the values of the dict.